pkg/ipamd/ipamd.go

```diff
@@ -601,21 +597,18 @@ func (c *IPAMContext) nodeInit() error {
 		c.askForTrunkENIIfNeeded(ctx)
 	}
 
-	if !c.disableENIProvisioning {
-		// For a new node, attach Cidrs (secondary ips/prefixes)
-		increasedPool, err := c.tryAssignCidrs()
-		if err == nil && increasedPool {
-			c.updateLastNodeIPPoolAction()
-		} else if err != nil {
-			if containsInsufficientCIDRsOrSubnetIPs(err) {
-				log.Errorf("Unable to attach IPs/Prefixes for the ENI, subnet doesn't seem to have enough IPs/Prefixes. Consider using new subnet or carve a reserved range using create-subnet-cidr-reservation")
-				c.lastInsufficientCidrError = time.Now()
-				return nil
-			}
-			return err
+	// On node init, check if datastore pool needs to be increased. If so, attach CIDRs from existing ENIs and attach new ENIs.
+	if !c.disableENIProvisioning && c.isDatastorePoolTooLow() {
+		c.increaseDatastorePool(ctx)
```

Comment:
We will have to capture the `err`  from `c.increaseDatastorePool `.. similar to - 

```
else if err != nil {
 			if containsInsufficientCIDRsOrSubnetIPs(err) {
 				log.Errorf("Unable to attach IPs/Prefixes for the ENI, subnet doesn't seem to have enough IPs/Prefixes. Consider using new subnet or carve a reserved range using create-subnet-cidr-reservation")
 				c.lastInsufficientCidrError = time.Now()
 				return nil
 			}
 			return err
```

Reply:
Ideally any issues while increasing the cidr during nodeInit we used to return `err` so have to maintain the same behavior..

Reply:
I considered that, but I want to be really careful in which cases we return error and crash IPAMD. For example, an EC2 API could fail temporarily, but then be resolved on the normal loop. While the pod would be restarted after the crash, it seems like we can allow a transient networking issue as long as we can make forward progress.

In the non-custom networking case, we can make forward progress since the primary ENI is already present. In the custom-networking case, we can make forward progress as long as at least one ENI was attached.

Reply:
But this will be a behavior change, previously we used to err out nodeInit and now nodeInit might pass through..

Reply:
That's a good point, I can catch this error and maintain the original behavior